libraries/chain/authorization_manager.cpp

```diff
@@ -308,31 +316,40 @@ namespace eosio { namespace chain {
    private:
       const authorization_manager& _authorization;
       vector<fc::microseconds> _max_delay_stack;
+      fc::microseconds _delay_max_limit;
       bool _track_delay;
    };
 
-   /**
-    * @param actions - the actions to check authorization across
-    * @param provided_keys - the set of public keys which have authorized the transaction
-    * @param allow_unused_signatures - true if method should not assert on unused signatures
-    * @param provided_accounts - the set of accounts which have authorized the transaction (presumed to be owner)
-    *
-    * @return fc::microseconds set to the max delay that this authorization requires to complete
-    */
-   fc::microseconds authorization_manager::check_authorization( const vector<action>& actions,
-                                                                const flat_set<public_key_type>& provided_keys,
-                                                                bool                             allow_unused_signatures,
-                                                                flat_set<account_name>           provided_accounts,
-                                                                flat_set<permission_level>       provided_levels
-                                                              )const
+   void noop_checktime( uint32_t ) {}
+
+   fc::microseconds
+   authorization_manager::check_authorization( const vector<action>&                actions,
+                                               const flat_set<public_key_type>&     provided_keys,
+                                               const flat_set<permission_level>&    provided_permissions,
+                                               fc::microseconds                     provided_delay,
+                                               const std::function<void(uint32_t)>& _checktime,
+                                               bool                                 allow_unused_keys
+                                             )const
    {
+      auto _noop_checktime = std::bind(&noop_checktime, std::placeholders::_1);
```

Comment:
Is it necessary for this to be a member function that is bound as opposed to a static/global `std::function<void(uint32_t)>` constructed with a simple lambda like `[](uint32_t){}`?